### ISSUE TITLE
[Meta] Migrate Libbeat plugins to the Global registry

### ISSUE THREAD

**Issue by ph** [elastic/beats#7423]
With the introduction of  #7304 we can refactor plugins to use the new registry instead of a custom registry for each plugin type.

- [ ] Autodiscover
  - [ ] Provider
  - [ ] builder
  - [ ] appenders
- [ ] Processor #7396 
- [ ] Queue
- [ ] Output #7400
- [ ] Codec (output codec)
- [ ] Refactor go-plugins to use the new _Feature_ instead of the old _Plugin_.

**Comment by exekias** [comment]
@urso is this one still relevant?

**Comment by urso** [comment]
Yes, it's still relevant. We might still need to evaulate if the solution in #7392 is good enough, but some ideas in it are definitely an improvement.

**Comment by jlind23** [comment]
Backlog grooming: Closing it for now until further activity, can still be reopened if needed.
